stan/math/rev/mat/fun/divide.hpp

```diff
@@ -9,18 +9,165 @@
 namespace stan {
 namespace math {
 
+namespace internal {
+template <int R, int C>
+class matrix_scalar_divide_vari_dv : public vari {
+ public:
+  int rows_;
+  int cols_;
+  vari* adjCRef_;
+  vari** adjResultRef_;
+  double invc;
+
+  explicit matrix_scalar_divide_vari_dv(const Eigen::Matrix<double, R, C>& m,
+                                        const var& c)
+      : vari(0.0),
+        rows_(m.rows()),
+        cols_(m.cols()),
+        adjCRef_(c.vi_),
+        adjResultRef_(ChainableStack::instance_->memalloc_.alloc_array<vari*>(
+            m.rows() * m.cols())) {
+    double c_d = c.val();
+    invc = 1.0 / c_d;
+    Eigen::Matrix<double, R, C> result = invc * m;
```

Comment:
[no action required]
This is a neat way to code this.  I'm guessing the matrix operation will more than make up for constructing the various matrices and maps and functors.

Reply:
Thanks. I adapted the pattern from the matrix-matrix multiply in multiply.hpp.